adapters/repos/db/vector/hnsw/distancer/asm/dot_byte_avx256.s

```diff
@@ -0,0 +1,182 @@
+//go:build !noasm && amd64
+// AUTO-GENERATED BY GOAT -- DO NOT EDIT
+
+TEXT ·dot_byte_256(SB), $0-32
+	MOVQ a+0(FP), DI
+	MOVQ b+8(FP), SI
+	MOVQ res+16(FP), DX
+	MOVQ len+24(FP), CX
+	BYTE $0x55               // pushq	%rbp
+	WORD $0x8948; BYTE $0xe5 // movq	%rsp, %rbp
+	BYTE $0x53               // pushq	%rbx
+	LONG $0xf8e48348         // andq	$-8, %rsp
+	WORD $0x8b4c; BYTE $0x11 // movq	(%rcx), %r10
+	WORD $0x8945; BYTE $0xd0 // movl	%r10d, %r8d
+	LONG $0x20fa8341         // cmpl	$32, %r10d
+	JGE  LBB0_1
+	WORD $0x8545; BYTE $0xd2 // testl	%r10d, %r10d
+	JLE  LBB0_7
+	LONG $0x10f88341         // cmpl	$16, %r8d
+	JAE  LBB0_10
+	WORD $0xc031             // xorl	%eax, %eax
+	WORD $0x3145; BYTE $0xc9 // xorl	%r9d, %r9d
+	JMP  LBB0_13
+
+LBB0_1:
+	WORD $0x634d; BYTE $0xc8 // movslq	%r8d, %r9
+	LONG $0xc0eff9c5         // vpxor	%xmm0, %xmm0, %xmm0
+	WORD $0xdb31             // xorl	%ebx, %ebx
+	WORD $0x894c; BYTE $0xc0 // movq	%r8, %rax
+
+LBB0_2:
+	WORD $0x8948; BYTE $0xd9       // movq	%rbx, %rcx
+	LONG $0x0c6ffec5; BYTE $0x1f   // vmovdqu	(%rdi,%rbx), %ymm1
+	LONG $0xe171edc5; BYTE $0x08   // vpsraw	$8, %ymm1, %ymm2
```

Comment:
@asdine is there a way for goat to generate the assembly mnemonics vs the hexadecimal values (especially given as the are added as comments) or is there a compatibility reason here?

Reply:
no, goat generates the assembly in hex by design, but I'm not sure why

Reply:
AFAIK this is a limitation of the plan9 assembly. Not all instructions exist in text form, but this way you can access them all.